src/SMap.hpp

```diff
@@ -18,17 +18,17 @@ struct SMap
     // Attributes //
     // Bitmaps for each field.
     // If n is a possible digit in field i, then 2^(n-1) & fields[i] is true.
-    uint16_t* fields;
-    bool* blank;
-    bool* fixed;
+    uint16_t fields[9*9];
+    bool blank[9*9];
+    bool fixed[9*9];
     int hash_value = 0;
 
     uint16_t sure_cand_row[9] = {};
     uint16_t sure_cand_col[9] = {};
 
     // saves  each straight
-    std::list<Straight*> row_straights[9];
-    std::list<Straight*> col_straights[9];
+    std::list<std::unique_ptr<Straight>> row_straights[9];
```

Comment:
I needed the header <memory> for this file to compile. Otherwise, unique_ptr is not known in namespace std.

Reply:
See next comment